tests/test_browser.go

```diff
@@ -423,3 +410,27 @@ func setupHTTPTestModuleInstance(tb testing.TB, samples chan k6metrics.SampleCon
 
 	return vu
 }
+
+func newBrowserTypeWithVU(tb testing.TB, opts *testBrowserOptions) (
+	_ *chromium.BrowserType,
+	_ *k6test.VU,
+	cancel func(),
+) {
+	tb.Helper()
+
+	vu := setupHTTPTestModuleInstance(tb, opts.samples)
+	registry := k6metrics.NewRegistry()
+	k6m := k6ext.RegisterCustomMetrics(registry)
+	vu.CtxField = k6ext.WithCustomMetrics(vu.Context(), k6m)
+	vu.InitEnvField.LookupEnv = opts.lookupFunc
+
+	ctx, cancel := context.WithCancel(vu.Context())
```

Comment:
`newBrowserTypeWithVU()` is like `context.WithCancel`; similar to returning a context and a cancel function. I thought `tb.Cleanup` makes the main `newTestBrowser()` clear that the browser stops when the test finishes. Otherwise, it becomes buried in `newBrowserTypeWithVU`, which can make it harder to see when it stops. Since `newTestBrowser` manages the lifecycle, also [here](https://github.com/grafana/xk6-browser/blob/0fc72a91b8ebaeeea92b624c48c2ea6b2b37d7df/tests/test_browser.go#L93-L101).

Reply:
Changed in f5dd191.